src/shared/proto/robot_component_status.proto

```diff
@@ -0,0 +1,122 @@
+syntax = "proto3";
+
+/* Data about the status of the break beam */
+message BreakBeamStatus
+{
+    // Whether or not the ball is interrupting the robot’s laser beam.
+    bool ball_in_beam = 1;
+
+    // The reading of the robot’s laser sensor, normalized to [0,1]
+    float break_beam_reading = 2;
+}
+
+/* Data about the status of the chipper and kicker */
+message ChipperKickerStatus
+{
+    // Indicates the time elapsed since chipper was last fired in ms
+    // UINT64_MAX indicates chipper was never fired
+    uint64 time_since_chipper_fired = 1;
+
+    // Indicates the time elapsed since kicker was last fired in ms
+    // UINT64_MAX indicates kicker was never fired
+    uint64 time_since_kicker_fired = 2;
+}
+
+/* Data about the status of the dribbler */
+message DribblerStatus
+{
+    int32 dribbler_rpm = 1;
+}
+
+/* Error codes are to be displayed in the visualizer */
+enum ErrorCode
+{
+    NO_ERROR                = 0;  // default message
+    ROBOT_DEAD_MESSAGE      = 1;
```

Comment:
`ROBOT_DEAD_MESSAGE` is a radio dongle-specific message. Perhaps it's worth putting any radio dongle errors in their own heading / group since they may be removed when radio support is dropped? (Just to make it easy to identify). This would go with things like
* `SECOND_DONGLE_MESSAGE`
* `ESTOP Broken`
* `TRANSMIT_QUEUE_FULL_MESSAGE`
etc.

Reply:
I don't think the complexity of another message isn't worth it. We can probably remove those enum values if we will never run old robots again. I added more comments.

Reply:
Yea sorry I didn't mean another message, just grouping in a comment. What you did looks good :+1: